modules/services/workspace-service/src/main/java/org/ballerinalang/composer/service/workspace/langserver/util/completion/filters/PackageActionAndFunctionFilter.java

```diff
@@ -41,55 +41,83 @@
 
         TokenStream tokenStream = dataModel.getTokenStream();
         int delimiterIndex = this.getPackageDelimeterTokenIndex(dataModel);
-
-        String searchTokenString = tokenStream.get(delimiterIndex - 1).getText();
         String delimiter = tokenStream.get(delimiterIndex).getText();
-        int colonTokenIndex;
-
-        if (delimiter.equals(":")) {
-            colonTokenIndex = delimiterIndex;
-        } else {
-            int initTokenIndex = this.getIndexOfTokenString(searchTokenString, 0, dataModel);
-            int equalTokenIndex = this.getIndexOfTokenString("create", initTokenIndex, dataModel);
-            colonTokenIndex = this.getIndexOfTokenString(":", equalTokenIndex, dataModel);
-        }
-
         ArrayList<SymbolInfo> returnSymbolsInfoList = new ArrayList<>();
 
-        if (colonTokenIndex > -1) {
-            String packageName = tokenStream.get(colonTokenIndex - 1).getText();
+        if (delimiter.equals(".")) {
```

Comment:
I would always use - if (".".equals(delimiter)) instead of - if (delimiter.equals(".")). 
It will avoid NPEs. (Eg: delimiter == null scenario)

Reply:
fixed